consensus/tendermint/tendermint.go

```diff
@@ -0,0 +1,637 @@
+package tendermint
+
+import (
+	"fmt"
+	"maps"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+type step uint
+
+const (
+	propose step = iota
+	prevote
+	precommit
+)
+
+type timeoutFn func(round uint) time.Duration
+
+type Addr interface {
+	// Ethereum Addresses are 20 bytes
+	~[20]byte | felt.Felt
+}
```

Comment:
Do you think it would be worth distinguishing between Ethereum and Starknet addresses / felts?

Reply:
If it is just through a wrapper than no because we should aim to represent type as close to the underlying type.